packages/synapse-react-client/src/components/Authentication/LastLoginInfo.tsx

```diff
@@ -0,0 +1,199 @@
+import { InfoTwoTone } from '@mui/icons-material'
+import { Box, Link, Tooltip, Typography } from '@mui/material'
+import React from 'react'
+import {
+  CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+} from '../../utils/SynapseConstants'
+import { LoginMethod, getLoginMethodFriendlyName } from './LoginMethod'
+
+export type LastLoginInfoProps = {
+  currentSourceAppName: string | null
+  lastLoginMethod: LoginMethod | null
+  lastLoginSourceAppName: string | null
+  lastLoginSourceAppURL: string | null
+  lastLoginDate: string | null
+  display: 'sentence' | 'box'
+}
+
+export function getLastLoginInfo(): Omit<LastLoginInfoProps, 'display'> {
+  return {
+    currentSourceAppName: localStorage.getItem(
+      CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginMethod: localStorage.getItem(
+      LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    ) as LoginMethod,
+    lastLoginSourceAppName: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginSourceAppURL: localStorage.getItem(
+      LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    ),
+    lastLoginDate: localStorage.getItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY),
+  }
+}
+
+export function setCurrentAppInfo(currentSourceAppName: string) {
+  localStorage.setItem(
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    currentSourceAppName,
+  )
+}
+
+export function setLastLoginInfo(
+  lastLoginMethod: LoginMethod,
+  lastLoginDate: string,
+  lastLoginSourceAppName: string,
+  lastLoginSourceAppURL: string,
+) {
+  localStorage.setItem(LAST_LOGIN_DATE_LOCALSTORAGE_KEY, lastLoginDate)
+  localStorage.setItem(LAST_LOGIN_METHOD_LOCALSTORAGE_KEY, lastLoginMethod)
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    lastLoginSourceAppName,
+  )
+  localStorage.setItem(
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    lastLoginSourceAppURL,
+  )
+}
+
+export function clearLastLoginInfo() {
+  const keys = [
+    CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  ]
+  keys.forEach(key => localStorage.removeItem(key))
+}
+
+function getSourceAppIndefiniteArticle(val: string) {
+  return val.match(/^[aeiou]/i) ? 'an' : 'a'
+}
```

Comment:
Used to handle portal names that start with vowels (e.g. ARK Portal) or consonants (e.g. Mobile Toolbox)